src/core/gate/garbling.rs

```diff
@@ -0,0 +1,185 @@
+use super::{GateId, GateType};
+use crate::{Delta, EvaluatedWire, GarbledWire, S};
+use digest::Digest;
+
+/// Generic hash function with unique tweak per gate using any digest implementation
+fn aes_hash<D: Digest + Default>(x: &S, tweak: GateId) -> S {
```

Comment:
I would suggest changing the name to generic_hash.

Reply:
Done!